packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2323,6 +2325,46 @@ export class ContainerRuntime
 		return result;
 	}
 
+	/**
+	 * Returns an aliased data store with the given alias.
+	 * @param alias - The alias for the data store.
+	 * @returns - The data store as a FluidObject if it exists and is aliased. Returns undefined if the data
+	 * store doesn't exists or is not aliased.
+	 */
+	public async getAliasedDataStore(alias: string): Promise<FluidObject | undefined> {
+		await this.dataStores.waitIfPendingAlias(alias);
+		const internalId = this.internalId(alias);
+		let context: FluidDataStoreContext;
+
+		try {
+			context = await this.dataStores.getDataStore(internalId, { wait: false });
```

Comment:
i really dislike error base control flow, and since getDataStore is a internal method we should just fix it here, or add an alternative that doesn't throw, and has the behavior we want, return undefined

Reply:
we're going to want to remove all the responseToException and createResponseError any way as we get ride of request